extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -444,4 +447,56 @@ public KafkaSupervisorTuningConfig getTuningConfig()
   {
     return spec.getTuningConfig();
   }
+
+  @Override
+  protected Map<KafkaTopicPartition, Long> getOffsetsFromMetadataStorage()
```

Comment:
Instead of overriding this method, you should just override the `checkSourceMetadataMatch` method.

Reply:
or rename this method appropriately so that callers know that its also filtering the spurious stored offsets

Reply:
Doesn't this fail when `updateDataSourceMetadataWithHandle` is called later on since that too will match the committed metadata with the new metadata?

Reply:
Updated logic for the KafkaDataSourceMetadata so that it handles topicPattern during the matching check in `updateDataSourceMetadataWithHandle`, thanks for pointing this out.